File: KWOC/KWOC/UseDemo/UIButtonExt_use/UIButtonExtShow.m
//
//  UIButtonExtShow.m
//  KWOC
//
//  Created by 渴望 on 2023/7/6.
//

#import "UIButtonExtShow.h"

@interface UIButtonExtShow ()

@end

@implementation UIButtonExtShow

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    UIButton *btn = [[UIButton alloc] initWithFrame:CGRectMake(20, 100, 300, 50)];
    [btn setBackgroundColor:[UIColor cyanColor]];
    [btn setTitle:@"默认样式" forState:UIControlStateNormal];
    [btn setTitleColor:[UIColor blackColor] forState:UIControlStateNormal];
    [btn setImage:[UIImage imageNamed:@"icon_wode02"] forState:UIControlStateNormal];
    [self.view addSubview:btn];

    
    UIButton *btn1 = [[UIButton alloc] initWithFrame:CGRectMake(20, 160, 300, 50)];
    [btn1 setBackgroundColor:[UIColor cyanColor]];
    [btn1 setTitle:@"图片在左侧 间距10" forState:UIControlStateNormal];
    [btn1 setTitleColor:[UIColor blackColor] forState:UIControlStateNormal];
    [btn1 setImage:[UIImage imageNamed:@"icon_wode02"] forState:UIControlStateNormal];
    [self.view addSubview:btn1];
    [btn1 kw_setWithImage:PositionLeft spacing:10];
    
    
    UIButton *btn2 = [[UIButton alloc] initWithFrame:CGRectMake(20, 220, 300, 50)];
    [btn2 setBackgroundColor:[UIColor cyanColor]];
    [btn2 setTitle:@"图片在右侧 间距10" forState:UIControlStateNormal];
    [btn2 setTitleColor:[UIColor blackColor] forState:UIControlStateNormal];
    [btn2 setImage:[UIImage imageNamed:@"icon_wode02"] forState:UIControlStateNormal];
    [self.view addSubview:btn2];
    [btn2 kw_setWithImage:PositionRight spacing:10];

    UIButton *btn3 = [[UIButton alloc] initWithFrame:CGRectMake(20, 280, 300, 100)];
    [btn3 setBackgroundColor:[UIColor cyanColor]];
    [btn3 setTitle:@"图片在上侧 间距10" forState:UIControlStateNormal];
    [btn3 setTitleColor:[UIColor blackColor] forState:UIControlStateNormal];
    [btn3 setImage:[UIImage imageNamed:@"icon_wode02"] forState:UIControlStateNormal];
    [self.view addSubview:btn3];
    [btn3 kw_setWithImage:PositionTop spacing:10];

    UIButton *btn4 = [[UIButton alloc] initWithFrame:CGRectMake(20, 390, 300, 100)];
    [btn4 setBackgroundColor:[UIColor cyanColor]];
    [btn4 setTitle:@"图片在下侧 间距10" forState:UIControlStateNormal];
    [btn4 setTitleColor:[UIColor blackColor] forState:UIControlStateNormal];
    [btn4 setImage:[UIImage imageNamed:@"icon_wode02"] forState:UIControlStateNormal];
    [self.view addSubview:btn4];
    [btn4 kw_setWithImage:PositionBottom spacing:10];
    
    
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: KW/KW/UseDemo/UIButtonExt_use/UIButtonExtShow.swift
//
//  UIButtonExtShow.swift
//  KW
//
//  Created by 渴望 on 2023/7/6.
//

import UIKit

class UIButtonExtShow: KSViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        
        
        
        let btn = UIButton(frame: CGRect(x: 20, y: 100, width: 300, height: 50))
        btn.backgroundColor = .cyan
        btn.setTitle("默认样式", for: .normal)
        btn.setTitleColor(.black, for: .normal)
        btn.setImage(UIImage(named: "icon_fabu"), for: .normal)
        view.addSubview(btn)
        
        
        let btn1 = UIButton(frame: CGRect(x: 20, y: 160, width: 300, height: 50))
        btn1.backgroundColor = .cyan
        btn1.setTitle("图片在左侧 间距10", for: .normal)
        btn1.setTitleColor(.black, for: .normal)
        btn1.setImage(UIImage(named: "icon_fabu"), for: .normal)
        view.addSubview(btn1)
        btn1.kw_set(image: .left, spacing: 10)
        
        
        let btn2 = UIButton(frame: CGRect(x: 20, y: 220, width: 300, height: 50))
        btn2.backgroundColor = .cyan
        btn2.setTitle("图片在右侧 间距10", for: .normal)
        btn2.setTitleColor(.black, for: .normal)
        btn2.setImage(UIImage(named: "icon_fabu"), for: .normal)
        view.addSubview(btn2)
        btn2.kw_set(image: .right, spacing: 10)
        
        
        let btn3 = UIButton(frame: CGRect(x: 20, y: 280, width: 300, height: 100))
        btn3.backgroundColor = .cyan
        btn3.setTitle("图片在上侧 间距10", for: .normal)
        btn3.setTitleColor(.black, for: .normal)
        btn3.setImage(UIImage(named: "icon_fabu"), for: .normal)
        view.addSubview(btn3)
        btn3.kw_set(image: .top, spacing: 10)
        
        
        let btn4 = UIButton(frame: CGRect(x: 20, y: 390, width: 300, height: 100))
        btn4.backgroundColor = .cyan
        btn4.setTitle("图片在下侧 间距10", for: .normal)
        btn4.setTitleColor(.black, for: .normal)
        btn4.setImage(UIImage(named: "icon_fabu"), for: .normal)
        view.addSubview(btn4)
        btn4.kw_set(image: .bottom, spacing: 10)
        
        
        
        
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destination.
        // Pass the selected object to the new view controller.
    }
    */

}
